tests/scripts/thread-cert/Cert_5_3_07_DuplicateAddress.py

```diff
@@ -117,7 +117,13 @@ def test(self):
         self.nodes[ED2].add_ipaddr('2001::1')
         time.sleep(3)
 
-        self.nodes[ED3].ping('2001::1')
+
+        # Final ping doesn't need a response;
+        # used to trigger an ADDR_ERR.ntf
+        try:
+            self.nodes[ED3].ping('2001::1')
+        except pexpect.TIMEOUT:
```

Comment:
It seems that doing a set-true/set-false of `PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE` after all the changes isn't sufficient, due to the following code in both `InsertPropertyHandler_THREAD_LOCAL_ROUTES()` and `InsertPropertyHandler_THREAD_ON_MESH_NETS()`, added in #273 (ping @abtink):

```
    VerifyOrExit(
        mAllowLocalNetworkDataChange == true,
        errorCode = SendLastStatus(header, SPINEL_STATUS_INVALID_STATE)
    );
```

This check is present to force the host implementation to use it in a way that is consistent with how we intended it to be used (Set this property to `true`, make your changes, then set it to `false`). I see three options to proceed forward:

1. Remove these checks from those functions in `ncp_base.cpp` and continue what you are doing.
2. Make the NCP automatically set `PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE` to `true` whenever a net data related property is modified.
3. Make sure your calls to add prefixes and local routes set `PROP_THREAD_ALLOW_LOCAL_NET_DATA_CHANGE` to true before setting the prefix. It is totally OK to set it to true multiple times.

I'm inclined to think option 3 would be the best since these checks force the host to interact with the NCP in the prescribed way—which should hopefully lead to better host implementations.